app/src/main/java/org/schabi/newpipe/info_list/holder/StreamMiniInfoItemHolder.java

```diff
@@ -111,19 +120,28 @@ public void updateState(final InfoItem infoItem,
                             final HistoryRecordManager historyRecordManager) {
         final StreamInfoItem item = (StreamInfoItem) infoItem;
 
-        final StreamStateEntity state = historyRecordManager
-                .loadStreamState(infoItem)
-                .blockingGet()[0];
+        StreamStateEntity state = null;
+        if (DependentPreferenceHelper.getPositionsInListsEnabled(itemProgressView.getContext())) {
+            state = historyRecordManager
+                    .loadStreamState(infoItem)
+                    .blockingGet()[0];
+        }
         if (state != null && item.getDuration() > 0
                 && !StreamTypeUtil.isLiveStream(item.getStreamType())) {
             itemProgressView.setMax((int) item.getDuration());
-            if (itemProgressView.getVisibility() == View.VISIBLE) {
-                itemProgressView.setProgressAnimated((int) TimeUnit.MILLISECONDS
-                        .toSeconds(state.getProgressMillis()));
+            if (DependentPreferenceHelper
+                    .getPositionsInListsEnabled(itemProgressView.getContext())) {
```

Comment:
This `if` condition is not needed, you can just run the code below directly. `state != null` is a condition of the `if` on line 129, and the state can be non-null only if positions in lists are enabled because of the `if` on line 124.